qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -404,3 +407,66 @@ def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **
         decomposition = synth_permutation_acg(high_level_object.pattern)
         return decomposition
+
+
+class TokenSwapperSynthesisPermutation(HighLevelSynthesisPlugin):
+    """The permutation synthesis plugin based on the token swapper algorithm.
+
+    This plugin name is :``permutation.token_swapper`` which can be used as the key on
+    an :class:`~.HLSConfig` object to use this method with :class:`~.HighLevelSynthesis`.
+
+    In more detail, this plugin is used to synthesize objects of type `PermutationGate`.
+    When synthesis succeeds, the plugin outputs a quantum circuit consisting only of swap
+    gates. When synthesis does not succeed, the plugin outputs `None`.
+
+    If either `coupling_map` or `qubits` is None, then the synthesized circuit
+    is not required to adhere to connectivity constraints, as is the case
+    when the synthesis is done before layout/routing.
+
+    On the other hand, if both `coupling_map` and `qubits` are specified, the synthesized
+    circuit is supposed to adhere to connectivity constraint. At the moment, the plugin
+    only works when `qubits` represents a connected subset of `coupling_map` (if this is
+    not the case, the plugin outputs `None`).
+
+    The plugin supports the following plugin-specific options:
+
+    * trials: The number of trials for the token swapper to perform the mapping. The
+      circuit with the smallest number of SWAPs is returned.
+
+    For more details on the token swapper algorithm, see to the paper:
+    `arXiv:1809.03452 <https://arxiv.org/abs/1809.03452>`__.
+
+    """
+
+    def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **options):
+        """Run synthesis for the given Permutation."""
+
+        pattern = high_level_object.pattern
+        trials = options.get("trials", 5)
+
+        pattern_as_dict = {j: i for i, j in enumerate(pattern)}
+
+        if coupling_map is None or qubits is None:
+            # The abstract synthesis uses a fully connected coupling map, allowing
+            # arbitrary connections between qubits
+            used_coupling_map = CouplingMap.from_full(len(pattern))
+        else:
+            # The concrete synthesis uses the coupling map restricted to the set of
+            # qubits over which the permutation gate is defined. Currently, we require
+            # this set to be connected (otherwise, replacing the node in the DAGCircuit
+            # that defines this PermutationGate by the DAG corresponding to the constructed
+            # decomposition becomes problematic); note that the method `reduce` raises an
+            # error if the reduced coupling map is not connected.
+            try:
+                used_coupling_map = coupling_map.reduce(qubits)
+            except CouplingError:
+                return None
+
+        graph = rx.PyGraph()
+        graph.extend_from_edge_list(list(used_coupling_map.get_edges()))
+        swapper = ApproximateTokenSwapper(graph, seed=1)
```

Comment:
Should `seed` ben an input option here? The other similar thought is the `parallel_threshold` argument to `ApproximateTokenSwapper.map` might also be worth exposing through the options interface, although that one is less clear to me because it's mostly just a function of runtime performance and not the actual synthesis quality.

Reply:
Thanks, I have exposed both ``seed`` and ``parallel_threshold``.